assets/src/dashboard/animations/animator/output.js

```diff
@@ -19,21 +19,45 @@
  */
 import PropTypes from 'prop-types';
 
-function WithAnimation({ id, style, children }) {
-  return (
-    <div id={id} style={style}>
+/**
+ * Internal dependencies
+ */
+import { ANIMATION_TYPE } from '../constants';
+
+function WithAnimation({
+  id,
+  animationType,
+  animationStyle,
+  containerStyle,
+  children,
+}) {
+  return animationType === ANIMATION_TYPE.REVEAL ? (
+    <div style={{ overflow: 'hidden', ...containerStyle }}>
+      <div id={id} style={{ width: '100%', height: '100%', ...animationStyle }}>
+        {children}
+      </div>
+    </div>
+  ) : (
```

Comment:
`Reveal` animations need to apply transform to a nested div with a parent that has `overflow: hidden;`

Reply:
@dvoytenko Yeah it looks like `clip-path` is _not_ allowed with `amp-animation`.  I just tried animating using it and I received a "Property is not whitelisted for animation" error.

However, if it could be added to `amp-animation` that'd be great because then I should be able to reproduce this same animation _without_ needing a container div by doing something like this:

```
keyframes: {
  transform: ['translateY(100%)', 'translateY(0%)'],
  clipPath: ['inset(0% 0% 100% 0%)', 'inset(0% 0% 0% 0%)'],
},
```
Combined it should appear as if the object were being uncovered as it traveled upwards (and it'd be just as easy to make left/right versions too).  This would give the same effect as it currently has without needing any parent div or overflow hidden or any of that extra stuff.  

In fact, there are a few other animations I've yet to tackle that would benefit from being able to animate using `clip-path`.  So, if it's doable to add to `amp-animation` relatively soon, please do and let me know when it's released so I can use it.

Reply:
Done: https://github.com/ampproject/amphtml/pull/27424. Should be in AMP PROD in two weeks or so.

Reply:
Woah, awesome!  I'll put this PR on hold until I can get the update and refactor `reveal` to use `clip-path` 👍